### PR TITLE
feat: add sorbet type checker

### PR DESCRIPTION
**Pull request by peterdeme**
- Add types. Runtime check is disabled, but static check is enabled.
- Add type checking to CI

### CODE REVIEW

File: lib/stream-chat/errors.rb [hunk 1]
```diff
@@ -1,25 +1,41 @@
...
+    # but for now, let's just run a static type check.
+
+    sig { returns(Integer) }
     attr_reader :error_code
+
+    sig { returns(String) }
     attr_reader :error_message
 
+    sig { returns(T::Boolean) }
+    attr_reader :json_response
```
Comment (ffenix113) [inline lib/stream-chat/errors.rb:19]: Should we keep `def json_response?` if we have `attr_reader`?

File: lib/stream-chat/types.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+# frozen_string_literal: true
+
+module StreamChat
+  extend T::Sig
+  T::Configuration.default_checked_level = :never
+  # For now we disable runtime type checks.
+  # We will enable it with a major bump in the future,
+  # but for now, let's just run a static type check.
+
+  StringKeyHash = T.type_alias { T::Hash[T.any(String, Symbol), T.untyped] }
```
Comment (ffenix113) [inline lib/stream-chat/types.rb:11]: We had issues with having strings/symbols in maps before, would you think we should allow only symbols here in next major version?

Comment (peterdeme) [reply lib/stream-chat/types.rb:11]: Agreed. Want me to modify this to this?

```
SymbolKeyHash = T.type_alias { T::Hash[Symbol, T.untyped] }
```

Comment (ffenix113) [reply lib/stream-chat/types.rb:11]: Yes, I think it would be right move.

File: lib/stream-chat/errors.rb (hunk 1 above)
Comment (peterdeme) [reply lib/stream-chat/errors.rb:19]: ✅

**Pull request merged**